sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala

```diff
@@ -1934,9 +1937,11 @@ abstract class CastBase extends UnaryExpression with TimeZoneAwareExpression wit
       case StringType =>
         val doubleStr = ctx.freshVariable("doubleStr", StringType)
         (c, evPrim, evNull) =>
+          val dt = ctx.addReferenceObj("doubleType", DoubleType, DoubleType.getClass.getName)
```

Comment:
This is overkill. Let's simply hardcode the data type name in the method `invalidInputSyntaxForNumericError` like https://github.com/apache/spark/pull/36244

Reply:
Sorry I forget to remove this line .. the `dt` is not used.